openl3/core.py

```diff
@@ -209,15 +214,327 @@ def process_file(filepath, output_dir=None, suffix=None, model=None,
+        # Get rid of alpha dimension
+        if image.shape[-1] == 4:
+            image = image[..., :3]
+
+    except Exception:
+        raise OpenL3Error('Could not open file "{}":\n{}'.format(filepath, traceback.format_exc()))
+
+    if not suffix:
+        suffix = ""
+
+    output_path = get_output_path(filepath, suffix + ".npz", output_dir=output_dir)
+
+    embedding = get_image_embedding(image, model=model,
+                                    input_repr=input_repr,
+                                    content_type=content_type,
+                                    embedding_size=embedding_size,
+                                    verbose=1 if verbose else 0)
+
+    np.savez(output_path, embedding=embedding)
+    assert os.path.exists(output_path)
+
+
+def process_video_file(filepath, output_dir=None, suffix=None,
+                       audio_model=None, image_model=None,
+                       input_repr="mel256", content_type="music",
+                       audio_embedding_size=6144, audio_center=True,
+                       audio_hop_size=0.1, image_embedding_size=8192,
+                       verbose=True):
+    """
+    Computes and saves L3 audio and video frame embeddings for given video file
+
+    Parameters
+    ----------
+    filepath : str
+        Path to video file to be processed.
+    output_dir : str or None
+        Path to directory for saving output files. If None, output files will
+        be saved to the directory containing the input file.
+    suffix : str or None
+        String to be appended to the output filename,
+        i.e. <base filename>_<modality>_<suffix>.npz.
+        If None, then no suffix will be added,
+        i.e. <base filename>_<modality>.npz.
+    audio_model : keras.models.Model or None
+        Loaded audio model object. If a model is provided, then `input_repr`,
+        `content_type`, and `embedding_size` will be ignored.
+        If None is provided, the model will be loaded using
+        the provided values of `input_repr`, `content_type` and
+        `embedding_size`.
+    image_model : keras.models.Model or None
+        Loaded audio model object. If a model is provided, then `input_repr`,
+        `content_type`, and `embedding_size` will be ignored.
+        If None is provided, the model will be loaded using
+        the provided values of `input_repr`, `content_type` and
+        `embedding_size`.
+    input_repr : "linear", "mel128", or "mel256"
+        Spectrogram representation used for audio model. Ignored if `model` is
+        a valid Keras model.
+    content_type : "music" or "env"
+        Type of content used to train embedding. Ignored if `model` is
+        a valid Keras model.
+    audio_embedding_size : 6144 or 512
+        Audio embedding dimensionality. Ignored if `model` is a valid Keras model.
+    audio_center : boolean
+        If True, pads beginning of audio signal so timestamps correspond
+        to center of window.
```

Comment:
As before, does this impact the alignment between audio/frame embeddings?

Reply:
Clarify in docs

Reply:
Will do.